File: demos/jans-chip/kmp/composeApp/src/androidMain/kotlin/services/AnalyticImpl.kt
package services

import com.example.fido2.services.Analytic
import com.example.fido2.services.Logger

class AnalyticImpl(private val logger: Logger): Analytic {
    override fun logEvent(event: String) {
        logger.log("Event \"$event\" sent to analytic by Android implementation")
    }
}

File: demos/jans-chip/kmp/iosApp/SuperGluuFido/Services/AnalyticImpl.swift
//
//  AnalyticImpl.swift
//  SuperGluuFido
//
//  Created by Nazar Yavornytskyi on 01.10.2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import shared

class AnalyticImpl: Analytic {
    private let logger: Logger
    
    init(logger: Logger) {
        self.logger = logger
    }
    
    func logEvent(event: String) {
        logger.log(text: "Event \"\(event)\" sent to analytic by iOS implementation")
    }
}
